src/drive/web/modules/drive/Toolbar/components/DocumentQualification.jsx

```diff
@@ -134,13 +134,19 @@ DocumentCategory.propTypes = {
  *
  */
 class DocumentQualification extends Component {
-  state = {
-    selected: {
-      categoryLabel: null,
-      itemId: null
+  constructor(props) {
+    super(props)
+    this.state = {
+      selected: {
+        categoryLabel:
+          props.selected && props.selected.categoryLabel
+            ? props.selected.categoryLabel
+            : null,
+        itemId:
+          props.selected && props.selected.itemId ? props.selected.itemId : null
```

Comment:
An update of the `props.selected` is not handled, therefore I think using initialSelected would be better.

Also, destructured-defaulting looks and reads nicer IMHO :)

```
const { categoryLabel = null, itemId = null } = props.selected || {}
this.state = { selected: { categoryLabel, itemId } }
```

Reply:
I don't want to handle the update, this is why this assign is in the constructor. Let's rename for initialSelected you right